File: app/src/main/java/com/example/steaminvestmentmanager/utilclasses/SettingsDialog.java
package com.example.steaminvestmentmanager.utilclasses;

import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Spinner;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import com.example.steaminvestmentmanager.MainActivity;
import com.example.steaminvestmentmanager.R;

import java.util.Objects;

public class SettingsDialog extends DialogFragment {
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.settings_dialog, null);
        Objects.requireNonNull(getDialog()).getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        Spinner currencySpinner = rootView.findViewById(R.id.currencySpinner);
        ArrayAdapter<String> currencyAdapter = new ArrayAdapter<>(getContext(), R.layout.my_simple_spinner_item, CurrencyData.getCurrencyArray());
        currencyAdapter.setDropDownViewResource(R.layout.my_simple_drop_down_item);
        currencySpinner.setAdapter(currencyAdapter);
        currencySpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                int newCurrency = CurrencyData.getCurrencyFromChar(CurrencyData.getCurrencyArray()[position]);
                CurrencyData.setCurrency(newCurrency);
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {
            }
        });
        return rootView;
    }
}

File: app/src/main/java/com/example/steaminvestmentmanager/utilclasses/SettingsDialog.kt
package com.example.steaminvestmentmanager.utilclasses

import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Spinner
import androidx.fragment.app.DialogFragment
import com.example.steaminvestmentmanager.R
import java.util.Objects

class SettingsDialog : DialogFragment() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.settings_dialog, null)
        Objects.requireNonNull(dialog)?.window!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        val currencySpinner = rootView.findViewById<Spinner>(R.id.currencySpinner)
        val currencyAdapter = context?.let { ArrayAdapter(it, R.layout.spinner_item, CurrencyData.currencyArray) }
        currencyAdapter?.setDropDownViewResource(R.layout.drop_down_item)
        currencySpinner.adapter = currencyAdapter
        currencySpinner.setSelection(CurrencyData.currencyArrayPosition)
        currencySpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(parent: AdapterView<*>?, view: View, position: Int, id: Long) {
                val newCurrency = CurrencyData.getCurrencyFromChar(CurrencyData.currencyArray[position])
                CurrencyData.setCurrency(newCurrency)
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {}
        }
        return rootView
    }
}